tests/functions.sh

```diff
@@ -257,38 +257,146 @@ messages {
   }
 }
 " > $f
-   
-    $BUTTERFLY_BUILD_ROOT/api/client/butterfly-client -e tcp://127.0.0.1:876$but_id -i $f
-    ret=$?
-    rm $f
-    if [ ! "$ret" == "0" ]; then
-        echo "client failed to send message to butterfly $but_id"
-        clean_all
-        exit 1
+
+    request $but_id $nic_id $f
+
+    if ! test -e /tmp/qemu-vhost-nic-$nic_id ; then
+	echo "client failed: we should have a socket in /tmp/qemu-vhost-nic-$nic_id"
+	clean_all
+	exit 1
     fi
+}
+
+function add_nic_void {
```

Comment:
Ok, I've just understand why you change "client_add_nic" to "add_nic_full_open"...
I was really thinking to refuse the patch "rename_funct" PR because renaming a function without a good reason is a really bad thing to do, splitting a function in 2 sub function can be a good reason to rename a function, and in this case it is.

maybe you can call this function "add nic" use "add_nic" in "add_nic_full_open", move this to the pool request "rename_funct".